src/sdk/namespace_fs.js

```diff
@@ -1964,70 +1980,43 @@ class NamespaceFS {
                 throw new S3Error(S3Error.InvalidObjectStorageClass);
             }
 
-            // Total 8 states
-            const restore_request = stat.xattr[GlacierBackend.XATTR_RESTORE_REQUEST] ?
-                parseInt(stat.xattr[GlacierBackend.XATTR_RESTORE_REQUEST], 10) : null;
-            const restore_ongoing = stat.xattr[GlacierBackend.XATTR_RESTORE_ONGOING] === 'true';
-            const restore_expiry = stat.xattr[GlacierBackend.XATTR_RESTORE_EXPIRY] ?
-                new Date(stat.xattr[GlacierBackend.XATTR_RESTORE_EXPIRY]) : null;
-
-            // 5 Valid States
-            if (restore_request) {
-                // It is possible that the restore request was made but we haven't yet
-                // started the restore process. Batching is an implementation detail
-                // and need not be leaked to the client.
-                if (!restore_ongoing && !restore_expiry) {
-                    dbg.log0('namespace_fs.restore_object: state - (restore_request, !restore_ongoing, !restore_expiry)');
-                    throw new S3Error(S3Error.RestoreAlreadyInProgress);
-                }
-            } else if (restore_ongoing) {
-                if (!restore_expiry) {
-                    dbg.log0('namespace_fs.restore_object: state - (!restore_request, restore_ongoing, !restore_expiry)');
-                    throw new S3Error(S3Error.RestoreAlreadyInProgress);
-                }
-            } else {
-                const now = new Date();
-
-                // The only entity that can remove the restore request is the underlying
-                // restore mechanism, implying that at some point it must have picked
-                // up the request to restore and finished the process.
-                if (restore_expiry && restore_expiry > now) {
-                    dbg.log0('namespace_fs.restore_object: state - (!restore_request, !restore_ongoing, restore_expiry > now)');
+            const now = new Date();
+            const restore_state = this._get_restore_status(stat.xattr, now, file_path);
```

Comment:
minor: the const restore_state name is confusing as it actually returns an object with a state inside, maybe call it just restore_status or just status?